services/course-information/src/main/java/edu/uga/devdogs/course_information/webscraping/Pdf.java

```diff
@@ -292,4 +292,36 @@ public static void downloadPDF(String uri, String fileLocation) throws IOExcepti
         fos.write(body.readAllBytes());
         fos.close();
     }
+
+    /**
+     * Extracts course codes and numbers from a PDF transcript.
+     *
+     * <p>This method reads a PDF file located at the specified file path, extracts the text content, and 
+     * parses lines to find course codes. A valid course code is determined by a set of criteria:
+     * it must be exactly four uppercase letters, followed by a course number that is exactly four 
+     * characters in length. The resulting list of course codes and numbers is returned as a 
+     * String array.
+     *
+     * @param fileLocation the location of the PDF transcript file.
+     * @return an array of Strings, where each entry is a course code and number in the format "CODE XXXX".
+     * @throws IOException if an error occurs while reading the PDF file.
+     */
+    public static String[] fromTranscript(String fileLocation) throws IOException {
```

Comment:
create another function that creates the pdf object from a byte object. we don't know how frontend/backend is passing us the unofficial pdf yet so it is better to account for both methods